test/powershell/Modules/Microsoft.PowerShell.Utility/JsonObject.Tests.ps1

```diff
@@ -0,0 +1,53 @@
+﻿# Unit tests for JsonObject
+Describe 'JsonObject' -tags "CI" {
+
+    function ShouldThrow
```

Comment:
It's totally fine to have custom assertions in your test. I think @iSazonov meant that the `try/catch` section in `ShouldThrow` should be better use the template, so it would be like
```
function ShouldThrow
{
   ....
   try {
       & $InputObject
       throw "Should throw ArgumentException"
    } catch {
       $_.FullyQualifiedErrorId | should be "ArgumentException"
    }
}
```

For those tests you found out, they were written before we came up with the guidelines, but we do want future test to follow the guidelines as much as possible 😄